pkg/proxy/proxy_test.go

```diff
@@ -425,135 +425,170 @@ func (s *TestProxySuite) TestProxy() {
 					},
 				}
 
+				rejectedHeaders := []headerToAdd{
+					{},
+					{"Impersonate-User", "myvalue"},
+					{"Impersonate-Group", "developers"},
+					{"Impersonate-Group", "admins"},
+					{"Impersonate-Extra-dn", "cn=jane,ou=engineers,dc=example,dc=com"},
+					{"Impersonate-Extra-acme.com%2Fproject", "some-project"},
+					{"Impersonate-Extra-scopes", "view"},
+					{"Impersonate-Extra-scopes", "development"},
+					{"Impersonate-Uid", "06f6ce97-e2c5-4ab8-7ba5-7654dd08d52b"},
+					{"Impersonate-New", "myvalue"},
+				}
+
 				for k, tc := range tests {
 					s.Run(k, func() {
 
-						// Test each request using both the default workspace URL and a URL that uses the
-						// workspace context. Both should yield the same results.
-						for workspaceContext, reqPath := range map[string]string{
-							"default workspace":    "http://localhost:8081/api/mycoolworkspace/pods",
-							"workspace context":    "http://localhost:8081/workspaces/mycoolworkspace/api/mycoolworkspace/pods",
-							"proxy plugin context": "http://localhost:8081/plugins/myplugin/workspaces/mycoolworkspace/api/mycoolworkspace/pods",
-						} {
-							s.Run(workspaceContext, func() {
-								// given
-								req, err := http.NewRequest(tc.ProxyRequestMethod, reqPath, nil)
-								require.NoError(s.T(), err)
-								require.NotNil(s.T(), req)
-
-								for hk, hv := range tc.ProxyRequestHeaders {
-									for _, v := range hv {
-										req.Header.Add(hk, v)
-									}
+						for _, firstHeader := range rejectedHeaders {
+							for _, additionalHeader := range rejectedHeaders {
```

Comment:
I'm confused, aren't `firstHeader` and `additionalHeader` the same here? Why do we add them twice?

Reply:
Talked to Matous, it's intentional so that we can test combinations of the rejectedHeaders